src/utils/detectors.js

```diff
@@ -14,13 +14,29 @@ const importLocalName = (name, state) => {
 }
 
 export const isStyled = (tag, state) => {
-  if (t.isCallExpression(tag) && t.isMemberExpression(tag.callee)) {
+  if (t.isCallExpression(tag) && t.isMemberExpression(tag.callee) && tag.callee.property.name !== 'default' /** ignore default for #93 below */) {
     // styled.something()
     return isStyled(tag.callee.object, state)
   } else {
     return (
       (t.isMemberExpression(tag) && tag.object.name === importLocalName('default', state)) ||
-      (t.isCallExpression(tag) && tag.callee.name === importLocalName('default', state))
+      (t.isCallExpression(tag) && tag.callee.name === importLocalName('default', state)) ||
+
+      /**
+       * Support Typescript import #93
```

Comment:
Small nit: I don't think we have to refer to the specific issue here. Also we do not need to refer to TypeScript at all. Multiple transpilers and other tools might change the import name. Even the user might change it on purpose and use `require()`. So I think we can keep the comment generic and say:

> styled-components might be imported using a `require()` call and assigned to a variable of any name.

The same we could add to `importLocalName` just using the default name

Reply:
I see the point, this issue cause from js require() itself not Typescript